src/OmniSharp.DotNetTest/Helpers/TestFeaturesDiscover.cs

```diff
@@ -0,0 +1,56 @@
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using OmniSharp.Abstractions.Services;
+
+namespace OmniSharp.DotNetTest.Helpers
+{
+    [Export(typeof(ISyntaxFeaturesDiscover))]
+    public class TestFeaturesDiscover : ISyntaxFeaturesDiscover
+    {
+        public bool NeedSemanticModel { get; } = true;
+
+        public IEnumerable<string> Discover(SyntaxNode node, SemanticModel semanticModel)
+        {
+            if (node is MethodDeclarationSyntax)
+            {
+                var method = node as MethodDeclarationSyntax;
+                if (IsTestMethod(method, semanticModel))
+                {
+                    var methodName = semanticModel.GetDeclaredSymbol(node).ToDisplayString();
+                    methodName = methodName.Substring(0, methodName.IndexOf('('));
+
+                    yield return $"XunitTestMethod:{methodName}";
```

Comment:
I guess my point is, `dotnet test` is inherently abstract, it just expects a tool to be available that will run tests.

My this is with adding this endpoint is once it's in, we can't drop it very easily.  Where it's easier to add a general one in, and extend that through `MEF` a little easier.

What makes `/rundotnettest` different than `/runtest`, besides the implementation.  I just don't want to see a world where we have `/rundotnettest` `/runmsbuildtest`, `/runfsharptest`, etc.